notebook/notebook/notebook.py

```diff
@@ -131,10 +171,153 @@ def external_url_for(path):
     return url + path
 
 
+def forbidden():
+    return 'Forbidden', 403
+
+
 @app.route('/healthcheck')
 def healthcheck():
     return '', 200
 
+##### New notebook methods meant for consumption from agnostic client#####
+###### Methods that do not call the auth-gateway /verify method must not be directly public #######
+@app.route('/verify/<svc_name>/', methods=['GET'])
+def validate_get(svc_name):
+    access_token = request.args.get('authorization')
+    token = request.args.get('token')
+
+    if not access_token:
+        return forbidden()
+
+    resp = requests.get(f'http://{AUTH_GATEWAY}/verify',
+                        headers={'Authorization': f'Bearer {access_token}'})
+
+    if resp.status_code != 200:
+        return forbidden()
+
+    user_id = resp.headers.get('User')
+
+    if not user_id:
+        return forbidden()
+
+    try:
+        mysql_conn = pymysql.connect(
+        host=IP, port=PORT, user=USER, passwd=PASS, db=DB)
+
+        with mysql_conn.cursor() as cur:
+            cur.execute(f"SELECT svc_name FROM sessions WHERE user_id='{user_id}' AND token='{token}'")
+
+            rows = cur.fetchall()
+
+            if len(rows) != 1 or rows[0][0] != svc_name:
+                return forbidden()
+
+            return '', 200
+    except:
+        log.exception("/validate : GET : failed")
+        return '', 500
+
+
+@app.route('/api', methods=['GET'])
+def new_api_get():
+    user_id = request.headers.get('User')
+
+    if not user_id:
+        return forbidden()
+
+    try:
+        mysql_conn = pymysql.connect(
+        host=IP, port=PORT, user=USER, passwd=PASS, db=DB)
+
+        sql = f"SELECT name, svc_name, token FROM {TABLE} WHERE user_id='{user_id}' AND state<>'{DELETED_STATE}'"
+
+        rows = []
+        with mysql_conn.cursor() as cur:
+            cur.execute(sql)
+            rows = cur.fetchall()
+
+        return jsonify(rows), 200
+    except Exception as e:
+        log.exception(e)
+        return '', 500
+
+
+@app.route('/api/<token>', methods=['DELETE'])
+def delete_notebook(token):
```

Comment:
A jupyter notebook's token is not a uniquely identifying piece of information. The unique identifier is either the pair of the pod name and the service name or the pod uuid which is generated by `start_pod`. If we need a uniquely identifying name for a pod-service pair we should start using the `uuid` generated by `start_pod`.

Reply:
I mean, it's a uuid v4, which is sufficiently unique to be used a primary key with low collision chance, and we have a 1:1 mapping between notebook and token. What isn't uniquely identifying? It is as uniquely identifying as the pod_id. 

In any case, all mysql stuff is gone in what i will push out tonight or tomorrow.

Reply:
Moreover, it requires an extra trip to the DB to get the pod_name and service_name. The pod_name and service_name are the unique identifiers of a jupyter notebook server, we should use those (and we should get rid of services anyway).

Reply:
There is no extra trip needed? A single lookup, whether db or local cache or some encrypted token, is always needed to verify that the jupyter notebook the user wants to access is actually one they own.

Here "jupyter_token" is a misnomer; we don't have any actual access token for notebooks, it's identical to an id in our case. I preferred an opaque string, but we can use svc_name or pod_name as you suggest.